bindings/java/src/main/com/apple/foundationdb/FutureKey.java

```diff
@@ -22,16 +22,27 @@
 
 import java.util.concurrent.Executor;
 
+import com.apple.foundationdb.EventKeeper.Events;
+
 class FutureKey extends NativeFuture<byte[]> {
-	FutureKey(long cPtr, Executor executor) {
+	private final EventKeeper eventKeeper;
+	FutureKey(long cPtr, Executor executor,EventKeeper eventKeeper) {
 		super(cPtr);
+		this.eventKeeper = eventKeeper;
 		registerMarshalCallback(executor);
 	}
 
 	@Override
 	protected byte[] getIfDone_internal(long cPtr) throws FDBException {
 		return FutureKey_get(cPtr);
 	}
+	
+	@Override
+	protected void postMarshal(byte[] value) {
```

Comment:
This needs to either call `close` or the superclass implementation of `postMarshal` in order to release the native resources.